### ISSUE TITLE
Verify that SAMHeaders are being stripped in all Spark tools once https://github.com/samtools/htsjdk/pull/327 is in 

### ISSUE THREAD

**Issue by droazen** [broadinstitute/gatk#1072]
Once https://github.com/samtools/htsjdk/pull/327 is merged into htsjdk and propagates to hellbender, let's audit our Spark tools to ensure that we are always serializing headerless SAMRecords

**Comment by droazen** [comment]
This is done -- I've added tests for `ReadsSparkSource` to verify that the headers are being stripped.
